.github/workflows/commit_artifacts.yml

```diff
@@ -322,17 +327,108 @@ jobs:
           grep -rl "$CURRENT_VERSION" ./compiled-rn || echo "No files found with $CURRENT_VERSION"
           grep -rl "$CURRENT_VERSION" ./compiled-rn | xargs -r sed -i -e "s/$CURRENT_VERSION/$LAST_VERSION/g"
           grep -rl "$CURRENT_VERSION" ./compiled-rn || echo "Version reverted"
-      - name: Check if only the REVISION file has changed
-        id: check_should_commit
+      - name: Check changes before signing
         run: |
           echo "Full git status"
+          git add .
           git status
           echo "===================="
-          echo "Checking for changes"
-          # Check if there are changes in the files other than REVISION or @generated headers
-          # We also filter out the file name lines with "---" and "+++".
-          if git diff -- . ':(exclude)*REVISION' | grep -vE "^(@@|diff|index|\-\-\-|\+\+\+|@generated SignedSource)" | grep "^[+-]" > /dev/null; then
+          if git status --porcelain | grep -qv '/REVISION'; then
+            echo "Changes detected"
+            echo "===== Changes ====="
+            git --no-pager diff -U0 --cached | grep '^[+-]' | head -n 50
+            echo "==================="
+          else
+            echo "No Changes detected"
+          fi
+      - name: Revert signatures
```

Comment:
So this is the clever bit. Once we've reverted the version strings, then we can re-generate the headers. If the only difference in the file was the version string, then this will output the same `@generated` header, and the working directory will be clean.